jbehave-core/src/main/java/org/jbehave/core/model/TableTransformers.java

```diff
@@ -204,4 +222,100 @@ public String transform(String tableAsString, TableParsers tableParsers, TablePr
             return tableAsString.replace(replacing, replacement);
         }
     }
+
+    public static class SelfReferencing implements TableTransformer {
+        private final ParameterControls parameterControls;
+        private final Pattern placeholderPattern;
+
+        public SelfReferencing(ParameterControls parameterControls) {
+            this.parameterControls = parameterControls;
+            placeholderPattern = Pattern.compile(
+                    parameterControls.nameDelimiterLeft() + "(.*?)" + parameterControls.nameDelimiterRight());
+        }
+
+        @Override
+        public String transform(String tableAsString, TableParsers tableParsers, TableProperties properties) {
+            String rowSeparator = properties.getRowSeparator();
+            TableRows rows = tableParsers.parseRows(tableAsString, properties);
+            List<String> headers = rows.getHeaders();
+
+            StringBuilder tableBuilder = new StringBuilder();
+            tableBuilder.append(createRowAsString(headers, properties.getHeaderSeparator())).append(rowSeparator);
+
+            for (List<String> row : resolveRows(getNamedRows(rows.getRows(), headers))) {
+                tableBuilder.append(
+                        createRowAsString(row, properties.getValueSeparator())).append(rowSeparator);
+            }
```

Comment:
https://github.com/jbehave/jbehave-core/blob/master/jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTable.java#L395-L404 - there is already very similar logic